sdk/appconfiguration/app-configuration/test/public/index.spec.ts

```diff
@@ -1140,4 +1155,249 @@ describe("AppConfigurationClient", () => {
       });
     });
   });
+
+  describe("FeatureFlag configuration setting", () => {
+    const clientFilters: (
+      | object
+      | FeatureFlagTargetingClientFilter
+      | FeatureFlagTimeWindowClientFilter
+      | FeatureFlagPercentageClientFilter
+    )[] = [
+      {
+        name: "Microsoft.TimeWindow",
+        parameters: {
+          start: "Wed, 01 May 2019 13:59:59 GMT",
+          end: "Mon, 01 July 2019 00:00:00 GMT"
+          // TODO: dates should accept `string | Date`
+          // Returning
+          //      - SDK would return Date for all the config-settings if parsable
+          //      - SDK would return strings for non parsable strings
+          // Passing as an argument
+          //      - If passed in a string, we'll pass it as is.
+          //      - If passed Date, we'll make it a string.
+        }
+      },
+      { name: "FilterX" },
+      {
+        name: "Microsoft.Targeting",
+        parameters: {
+          audience: {
+            groups: [
+              { name: "group-1", rolloutPercentage: 25 },
+              { name: "group-2", rolloutPercentage: 45 }
+            ],
+            users: ["userA", "userB"],
+            defaultRolloutPercentage: 40
+          }
+        }
+      },
+      { name: "Microsoft.Percentage", parameters: { value: 25 } }
+    ];
+
+    const baseSetting: FeatureFlag = {
+      conditions: {
+        clientFilters
+      },
+      enabled: false,
+      isReadOnly: false,
+      key: `${featureFlagPrefix + generateUuid()}`,
+      contentType: featureFlagContentType,
+      description: "I'm a description",
+      label: "label-1"
+    };
+
+    let addResponse: AddConfigurationSettingResponse;
+
+    beforeEach(async () => {
+      addResponse = await client.addConfigurationSetting(baseSetting);
+    });
+
+    afterEach(async () => {
+      await client.deleteConfigurationSetting({
+        key: baseSetting.key,
+        label: baseSetting.label
+      });
+    });
+
+    function assertFeatureFlagProps(actual: FeatureFlag, expected: FeatureFlag) {
+      assert.equal(
+        actual.key,
+        expected.key,
+        "Key from the response from get request is not as expected"
+      );
+      assert.deepEqual(
+        actual.conditions,
+        expected.conditions,
+        "conditions from the response from get request is not as expected"
+      );
+      assert.equal(actual.description, expected.description);
+      assert.equal(actual.enabled, expected.enabled);
+      assert.equal(actual.isReadOnly, expected.isReadOnly);
+      assert.equal(actual.label, expected.label);
+      assert.equal(actual.contentType, expected.contentType);
+    }
+
+    it("can add and get FeatureFlag", async () => {
+      assert.equal(isFeatureFlag(addResponse), true, "Expected to get the feature flag");
+      if (isFeatureFlag(addResponse)) {
```

Comment:
One key thing is that the Type inference can only be leveraged by the users only inside the if-block.